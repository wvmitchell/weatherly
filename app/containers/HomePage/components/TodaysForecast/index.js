/**
*
* TodaysForecast
*
*/

import React from 'react';
import {
  Card,
  CardHeader,
  CardText,
} from 'material-ui/Card';


class TodaysForecast extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <Card className="todays-weather">
        <CardHeader
          title={`${this.props.conditions.display_location.full} - ${this.props.forecast.simple.forecastday[0].date.pretty}`}
          subtitle={this.props.getTempExtremesString(this.props.forecast, 0)}
          avatar={this.props.forecast.txt.forecastday[0].icon_url}
        />
        <CardText>
          <h3>Currently</h3>
          <p>{this.props.conditions.weather}, {this.props.conditions.temperature_string}</p>
          <br />
          <h3>Today</h3>
          <p>{this.props.forecast.txt.forecastday[0].fcttext}</p>
          <br />
          <h3>Tonight</h3>
          <p>{this.props.forecast.txt.forecastday[1].fcttext}</p>
        </CardText>
      </Card>
    );
  }
}

TodaysForecast.propTypes = {
  conditions: React.PropTypes.object,
  forecast: React.PropTypes.object,
  getTempExtremesString: React.PropTypes.func,
};

export default TodaysForecast;
