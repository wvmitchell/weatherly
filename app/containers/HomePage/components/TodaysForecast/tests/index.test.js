import React from 'react';
import { shallow } from 'enzyme';

import TodaysForecast from '../index';

describe('<TodaysForecast />', () => {
  it('should match the snapshot', () => {
    const mockConditions = {
      display_location: {
        full: 'some-place',
      },
      weather: 'some-weather',
      temperature_string: 'some-temp',
    };
    const mockForecast = {
      simple: {
        forecastday: [
          { date: { pretty: 'a date' } },
        ],
      },
      txt: {
        forecastday: [
          { fcttext: 'description' },
          { fcttext: 'description' },
        ],
      },
    };
    const renderedComponent = shallow(
      <TodaysForecast
        conditions={mockConditions}
        forecast={mockForecast}
        getTempExtremesString={jest.fn()}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
