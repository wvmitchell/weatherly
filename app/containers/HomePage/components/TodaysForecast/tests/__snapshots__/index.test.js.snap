exports[`<TodaysForecast /> should match the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Card
    className="todays-weather"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}>
    <CardHeader
        avatar={null}
        subtitle={undefined}
        title="some-place - a date" />
    <CardText>
        <h3>
            Currently
        </h3>
        <p>
            some-weather
            , 
            some-temp
        </p>
        <br />
        <h3>
            Today
        </h3>
        <p>
            description
        </p>
        <br />
        <h3>
            Tonight
        </h3>
        <p>
            description
        </p>
    </CardText>
</Card>,
  "nodes": Array [
    <Card
      className="todays-weather"
      expandable={false}
      expanded={null}
      initiallyExpanded={false}>
      <CardHeader
            avatar={null}
            subtitle={undefined}
            title="some-place - a date" />
      <CardText>
            <h3>
                  Currently
            </h3>
            <p>
                  some-weather
                  , 
                  some-temp
            </p>
            <br />
            <h3>
                  Today
            </h3>
            <p>
                  description
            </p>
            <br />
            <h3>
                  Tonight
            </h3>
            <p>
                  description
            </p>
      </CardText>
</Card>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <TodaysForecast
        conditions={
                Object {
                        "display_location": Object {
                          "full": "some-place",
                        },
                        "temperature_string": "some-temp",
                        "weather": "some-weather",
                      }
        }
        forecast={
                Object {
                        "simple": Object {
                          "forecastday": Array [
                            Object {
                              "date": Object {
                                "pretty": "a date",
                              },
                            },
                          ],
                        },
                        "txt": Object {
                          "forecastday": Array [
                            Object {
                              "fcttext": "description",
                            },
                            Object {
                              "fcttext": "description",
                            },
                          ],
                        },
                      }
        }
        getTempExtremesString={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TodaysForecast {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "conditions": Object {
            "display_location": Object {
              "full": "some-place",
            },
            "temperature_string": "some-temp",
            "weather": "some-weather",
          },
          "forecast": Object {
            "simple": Object {
              "forecastday": Array [
                Object {
                  "date": Object {
                    "pretty": "a date",
                  },
                },
              ],
            },
            "txt": Object {
              "forecastday": Array [
                Object {
                  "fcttext": "description",
                },
                Object {
                  "fcttext": "description",
                },
              ],
            },
          },
          "getTempExtremesString": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Card
          className="todays-weather"
          expandable={false}
          expanded={null}
          initiallyExpanded={false}>
          <CardHeader
                    avatar={null}
                    subtitle={undefined}
                    title="some-place - a date" />
          <CardText>
                    <h3>
                              Currently
                    </h3>
                    <p>
                              some-weather
                              , 
                              some-temp
                    </p>
                    <br />
                    <h3>
                              Today
                    </h3>
                    <p>
                              description
                    </p>
                    <br />
                    <h3>
                              Tonight
                    </h3>
                    <p>
                              description
                    </p>
          </CardText>
</Card>,
        "_debugID": 2,
        "_renderedOutput": <Card
          className="todays-weather"
          expandable={false}
          expanded={null}
          initiallyExpanded={false}>
          <CardHeader
                    avatar={null}
                    subtitle={undefined}
                    title="some-place - a date" />
          <CardText>
                    <h3>
                              Currently
                    </h3>
                    <p>
                              some-weather
                              , 
                              some-temp
                    </p>
                    <br />
                    <h3>
                              Today
                    </h3>
                    <p>
                              description
                    </p>
                    <br />
                    <h3>
                              Tonight
                    </h3>
                    <p>
                              description
                    </p>
          </CardText>
</Card>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TodaysForecast
    conditions={
        Object {
            "display_location": Object {
              "full": "some-place",
            },
            "temperature_string": "some-temp",
            "weather": "some-weather",
          }
    }
    forecast={
        Object {
            "simple": Object {
              "forecastday": Array [
                Object {
                  "date": Object {
                    "pretty": "a date",
                  },
                },
              ],
            },
            "txt": Object {
              "forecastday": Array [
                Object {
                  "fcttext": "description",
                },
                Object {
                  "fcttext": "description",
                },
              ],
            },
          }
    }
    getTempExtremesString={[Function]} />,
}
`;
